### DISCUSSION TITLE
不在 git 中跟踪 .vscode 文件夹

### DISCUSSION THREAD

**Discussion by timongh** [the1812/Bilibili-Evolved#3202] [功能建议 / Ideas]
现目前这个项目在 git 中跟踪了 .vscode 文件夹。这样带来的问题是：其他开发者难以自定义自己私人的配置。各开发者的本地环境并不相同，一套配置难以满足所有人的需求。\[注\]

我的想法是，停止 git 对 .vscode 文件夹的跟踪。然后将现有的 .vscode 文件夹改名为 vscode-example、vscode-recommended 等。这样所有开发者都能自行选择用还是不用。

（我本地目前用的是 WebStorm，我把 tasks.json 改成了 WebStorm 中对应的运行配置文件。如果上述提议可行的话，我把我的运行文件用同样的方式传上来。）

\[注\]：可以用 `git update-index --skip-worktree` 绕过

**Comment by the1812** [comment]
私人的配置应该是在 vscode 整体的 settings.json 里的, 项目的 .vscode/settings.json 只是用来预置一些项目相关的配置, 比如忽略编译产物的 js 高亮, 定义一些自定义词语给拼写检查等; .vscode 下其他的文件也和私人配置无关.
我不知道 WebStorm 的机制是怎么样的, 应该也有全局配置的吧, 只把项目特定配置传上来就可以.

**Reply by timongh** [reply]
我举几个例子，我之前用的也是vscode。

1. 我想修改 tasks.json。比如 lint 不带 --fix（只是例子）
2. 再比如我在另一个讨论里提的，lint 不想要 html 输出。那么在 github 代码修改前，我自己就能先改掉。
3. 假如我本地只给 npx 配了环境变量，yarn, eslint, tsc 等都需要通过 npx 间接调用，那我就需要修改 tasks.json 才能在本地使用
4. setting.json 里面配置的格式化程序是 vscode 自带格式化程序。而我想换成 eslint 或者 prettier

而且私人配置也不总是在全局配置。仅在单个项目启用的私人配置也挺常见的。

**Reply by the1812** [reply]
1~3 都是 tasks.json 的问题, tasks.json 提供的就是项目的一些基础开发配置, 使用前提是 yarn 全局可用, eslint / tsc 这些都是由 yarn 唤起. 如果对 task 有改进想法, 也应该像其他代码一样提交到 GitHub 来.
4. 格式化方面, 目前把这个放在项目的 settings.json 里确实不妥, 应该去掉.

**Reply by FoundTheWOUT** [reply]
格式化我觉得还是统一一下比较好。但至于放不放 settings.json 也许都可以，只要搞个 githook（比如加个 husky），在提交时一并检查和格式化就好了

**Reply by the1812** [reply]
因为我还没把 prettier 配置合并到 eslint 里, 最后会统一用 eslint 的.
检测时机方面, 代码库本身 eslint 全通过 + IDE 集成的提示 + PR 和 master 下的 CI 检查, 完全够用了, 没必要在 commit 上卡点浪费时间.
